epochStart/metachain/legacySystemSCs.go

```diff
@@ -1060,41 +1031,7 @@ func (s *legacySystemSCProcessor) callSetOwnersOnAddresses(arguments [][]byte) e
 }
 
 func (s *legacySystemSCProcessor) initDelegationSystemSC() error {
-	codeMetaData := &vmcommon.CodeMetadata{
-		Upgradeable: false,
-		Payable:     false,
-		Readable:    true,
-	}
-
-	vmInput := &vmcommon.ContractCreateInput{
-		VMInput: vmcommon.VMInput{
-			CallerAddr: vm.DelegationManagerSCAddress,
-			Arguments:  [][]byte{},
-			CallValue:  big.NewInt(0),
-		},
-		ContractCode:         vm.DelegationManagerSCAddress,
-		ContractCodeMetadata: codeMetaData.ToBytes(),
-	}
-
-	vmOutput, err := s.systemVM.RunSmartContractCreate(vmInput)
-	if err != nil {
-		return err
-	}
-	if vmOutput.ReturnCode != vmcommon.Ok {
-		return epochStart.ErrCouldNotInitDelegationSystemSC
-	}
-
-	err = s.processSCOutputAccounts(vmOutput)
-	if err != nil {
-		return err
-	}
-
-	err = genesisCommon.UpdateSystemSCContractsCode(vmInput.ContractCodeMetadata, s.userAccountsDB)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return genesisCommon.InitDelegationSystemSC(s.systemVM, s.userAccountsDB)
```

Comment:
the legacy files should not be touched. Even if it means copy paste code.

Reply:
Reverted the code